Samples/ThreadedResourcePrep/src/ThreadedResourcePrep.cpp

```diff
@@ -0,0 +1,356 @@
+    defineSelectableMesh("dragon.mesh",Vector3(75,28,-25), Vector3(0.2, 0.2, 0.2), Degree(203));
+    defineSelectableMesh("razor.mesh",Vector3(-15,17,10), Vector3(0.1, 0.1, 0.1), Degree(-16));
+    // these have no material but are big - good for the benchmark
+    defineSelectableMesh("geosphere4500.mesh",Vector3(40,0,-25), Vector3(0.01, 0.01, 0.01), Degree(180));
+    defineSelectableMesh("geosphere8000.mesh",Vector3(-40,0,-25), Vector3(0.01, 0.01, 0.01), Degree(180));
+}
+
+void Sample_ThreadedResourcePrep::setupControls()
+{
+    mTrayMgr->showCursor();
+
+    const float BOTTOM_W = 370;
+    const float TOPLEFT_W = 200;
+
+    // create a menu to choose the model displayed
+    mMeshMenu = mTrayMgr->createLongSelectMenu(TL_BOTTOM, "Mesh", "Mesh", BOTTOM_W, 290, 10);
+
+    mMeshStatLabel = mTrayMgr->createLabel(TL_BOTTOM, "MeshStat", "", BOTTOM_W);
+    refreshStatsUi();
+
+    mReloadBtn = mTrayMgr->createButton(TL_BOTTOM, "ReloadBtn", "Reload!", BOTTOM_W/2);
+
+    mBatchSlider = mTrayMgr->createThickSlider(TL_TOPLEFT, "BatchSlider", "Batch length", TOPLEFT_W, 50, 1, (Real)mMeshQueue.size(), (int)mMeshQueue.size());
+
+    mThreadedMeshChk = mTrayMgr->createCheckBox(TL_TOPLEFT, "ThrMeshChk", "Bg. mesh prep", TOPLEFT_W);
+
+    mUnloadBtn = mTrayMgr->createButton(TL_TOPLEFT, "UnloadBtn", "Unload batch", TOPLEFT_W);
+
+    mBatchSlider->setValue((Real)mMeshQueue.size(), /* notifyListener: */false);
+    mBatchSize = mMeshQueue.size();
+
+    refreshMeshUi();
+    mMeshMenu->selectItem(0, /* notifyListener: */false);
+}
+
+void Sample_ThreadedResourcePrep::itemSelected(SelectMenu* menu)
+{
+    if (menu == mMeshMenu)
+    {
+        // Find selected mesh by name (account for "<in bulk>" in name)
+        StringVector toks = Ogre::StringUtil::split(mMeshMenu->getSelectedItem());
+        auto selectedMeshItor = std::find_if (mMeshQueue.begin(), mMeshQueue.end(),
+            [&toks](MeshInfo& mi)
+            {
+                return mi.meshName == toks[0];
+            });
+
+        // focus camera
+        SceneNode* camnode = mCameraMan->getCamera();
+        camnode->lookAt(selectedMeshItor->sceneNode->getPosition(), Node::TS_WORLD);
+
+        // Move selected item to front of queue
+        MeshInfo mi = *selectedMeshItor;
+        mMeshQueue.erase(selectedMeshItor);
+        mMeshQueue.push_front(mi);
+        refreshMeshUi();
+
+        mStats.resetStats();
+        refreshStatsUi();
+    }
+}
+
+void Sample_ThreadedResourcePrep::sliderMoved(Slider* slider)
+{
+    if (slider == mBatchSlider)
+    {
+        size_t bulkSize = (size_t)mBatchSlider->getValue();
+        if (bulkSize != mBatchSize)
+        {
+            mBatchSize = bulkSize;
+            refreshMeshUi();
+            mStats.resetStats();
+            refreshStatsUi();
+        }
+    }
+}
+
+void Sample_ThreadedResourcePrep::refreshStatsUi()
+{
+    std::stringstream buf;
+    buf << "Reloads: " << mStats.totalReloads 
```

Comment:
its better to use `StringUtil::format`

https://ogrecave.github.io/ogre/api/latest/class_ogre_1_1_string_util.html#ac01104ece71cfc3f7a951ed031518125